src/fsharp/FSharp.Core/list.fs

```diff
@@ -356,12 +352,10 @@ namespace Microsoft.FSharp.Collections
         let exists predicate list = Microsoft.FSharp.Primitives.Basics.List.exists predicate list
         
         [<CompiledName("Contains")>]
-        let inline contains value source =
-            let rec contains e xs1 =
-                match xs1 with
-                | [] -> false
-                | h1::t1 -> e = h1 || contains e t1
-            contains value source
+        let rec contains value source =
```

Comment:
Does it matter that contains is no longer marked as `inline`?

Reply:
@isaacabraham idk, but neither are `exists`, `find/findBack`, `tryFind/tryFindBack`, `pick/tryPick` etc.

Reply:
Hmmm - so why was `contains` inlined? Just some historical artifact, or some reason.

Reply:
People started to inline stuff in other PRs...

Reply:
Yes, the "inline" matters a lot.  The inlining causes the equality check to become monomorphic (non-generic) in the common case that the callsite is monmorphic, and hence the equality check will be type-specialized

Reply:
@dsyme I'll take it out if the inline is that important, as I'm not sure how to get around FS1118 (cannot inline recursive).
There are a few other funcs with `when 'T : equality` signature (except, distinct, countBy etc.), should they be inline as well?

Reply:
Yes, the nested ``let rec`` is to get around the restriction on ``let rec inline``.  The whole inner ``let rec`` gets copied that way.

Was the use of a nested ``let rec`` causing problems for Fable?  If so you may often be able to replace with an explicit loop

Reply:
@dsyme Nothing to do with Fable, this optimization was just to remove the unnecessary extra object allocation and call that is generated, but I understand the inline modifier is more important here (although there are other funcs with `when 'T : equality` that are not inline).

Reply:
> I understand the inline modifier is more important here (although there are other funcs with when 'T : equality that are not inline).

If there is  performance evidence to show that it's worthwhile inlining these then we should likely do it.  Normally it's easy to get perf evidence in such cases. Sometimes the inlining can't be done due to the implementation depending on other internal functions we don't want to make public